### PR TITLE
fix(cli): codegen — defineModules chain detection, sibling versions, agent docs

### PR DESCRIPTION
**Pull request by forinda**
## Why

Adopters running fresh `kick new` projects hit five distinct papercuts. This PR addresses all of them in one focused patch.

### 1. `kick g module` now extends `defineModules()` chains

The orchestrator's array-insertion regex only matched flat `[...]` literals. Projects whose `src/modules/index.ts` used the fluent `defineModules().mount(...)` form (now the default) saw new modules' import lines added but `.mount(NewModule())` missing — the new module silently never registered.

Fix: scanner detects both shapes. Flat array stays on the existing path; fluent chain gets a balanced-paren walker that handles nested factory calls (`mount(UserModule())`) without the inner parens breaking boundary detection. Anchor on `defineModules\s*\(` so the import statement isn't matched first.

### 2. New projects default to `defineModules()`

`kick new` and `kick g module` on a fresh project now emit:

```ts
import { defineModules } from '@forinda/kickjs'
import { HelloModule } from './hello/hello.module'

export const modules = defineModules().mount(HelloModule())
```

Subsequent `kick g module <name>` invocations append `.mount(<Name>Module())` to the chain. Pinning `modules.style: 'class'` in `kick.config.ts` keeps the legacy flat-array form.

### 3. `kick new` resolves each sibling's actual published version

`kick new` previously pinned every `@forinda/kickjs-*` dep to the CLI's own version. After per-package independent versioning landed, that under-installs adopters whenever a sibling bumps independently — `@forinda/kickjs@5.5.0` may pair with `@forinda/kickjs-cli@5.4.2` and `@forinda/kickjs-swagger@5.3.1`.

Fix: parallel `npm view <name> version` per sibling at scaffold time, via `execFileSync` (no shell injection), 5s timeout each. Failure (offline / unpublished) falls back to the CLI version. Bonus: scaffolded `package.json` starts at `0.0.0` instead of inheriting the CLI version (which broke adopters' first npm publish).

### 4. Drop `buildRoutes()` mechanics from generated `routes()` JSDoc

The generated `routes()` JSDoc lectured adopters on how the framework derives the Express Router from the controller via `buildRoutes()` — implementation detail, not API documentation. Replaced with a focused breakdown of the return-value shape (`path` / `controller` / `version`).

### 5. Agent docs cover the new module API

Generated `CLAUDE.md` / `AGENTS.md` / `kickjs-skills.md` now describe `defineModule({...})` + `defineModules().mount(...)` as the default module shape, name `kick.config.ts > modules.style: 'define' | 'class'` as the toggle, and point at `kick codemod modules --experimental --apply` for migrating between the two forms. Cheat-sheet rows updated; registry-array snippets switched to the fluent chain; `AppModule` interface reframed as the legacy alternative.

## Test plan

- [x] `pnpm test` (CLI) — 257/257 pass (1 existing test updated for the new default; 1 new regression test for chain-append on fluent-form registries)
- [x] `pnpm typecheck` (CLI) — clean
- [x] `pnpm build` (CLI) — clean
- [x] Manual repro: ran `kick g module tiger` against the user's `kick-beta/my-app` (a project with the existing `defineModules()` chain) — `.mount(TigerModule())` correctly appended

## Reproducing the original bug

The user's `kick-beta/my-app` had a `defineModules().mount(HelloModule()).mount(BuffaloModule()).mount(...)` chain. Running `kick g module lion` against it added `import { LionModule } ...` but skipped `.mount(LionModule())` — the regex `(=\s*\[)([\s\S]*?)(])` matched nothing in fluent-chain syntax. After this PR, the same command appends `.mount(LionModule())` to the end of the chain.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Scaffolded projects now default to a fluent defineModules().mount(...) registry and append mounts when adding modules
  * Scaffolding resolves sibling package versions from published releases (fallback applied) and sets generated project version to 0.0.0
  * New scaffold options for module declaration style and token scope (defaults provided)

* **Bug Fixes**
  * Robust add/remove behavior for chained/nested .mount(...) registries and safer import/identifier handling
  * CLI now honors global options (e.g., --force) when generating modules

* **Documentation**
  * Migration guidance, cheat-sheet snippets, and routes JSDoc updated for the new module API

* **Tests**
  * E2E tests expanded to cover fluent-chain append/remove, collision and comment-edge cases

* **Chores**
  * .claude/ added to .gitignore

### CODE REVIEW

**Comment by forinda** [comment]
Thanks @coderabbitai — addressed in [`f176415`](https://github.com/forinda/kick-js/pull/196/commits/f1764154).

Hoisted `appendModuleEntry` to an exported helper on `generators/module.ts` and called it from `scaffold.ts`'s `autoRegisterModule`. Single source of truth for shape detection + balanced-paren walk; the scaffold path inherits the chain support that the orchestrator gained in #196 instead of duplicating (and now-fixing) the same regex twice.

Side note on the `npm view` parallelism nit — agreed it's lower-impact (scaffolding is one-shot, npm queries usually return <1s, fallback fires fast on offline). Filing as a follow-up rather than expanding this PR; happy to land it next session if you'd prefer.

**Comment by forinda** [comment]
Both addressed in [`5cd1395`](https://github.com/forinda/kick-js/pull/196/commits/5cd1395b).

**1. Repetition** — `autoRegisterModule` lived in both `module.ts` and `scaffold.ts` line-for-line; the only reason the two diverged was the regex bug from earlier in this PR (the one you flagged on f176415). Hoisted to a single exported function on `module.ts`; scaffold imports + calls it. As a side effect `scaffold.ts` no longer needs `readFile` / `writeFile` / `fileExists` — cleaned those imports up too.

**2. `.claude/settings.local.json`** — removed from git, added `.claude/` to `.gitignore`. Re-commitment is blocked going forward. The overbroad `Bash(pnpm remove *)` permission inside that file is moot since the file no longer ships in the repo; my local copy stays my problem.

257/257 still passing.

File: packages/cli/src/generators/module.ts [hunk 1]
```diff
@@ -185,21 +206,128 @@ export const modules: AppModuleEntry[] = [${entryToken}]
...
-  await writeFile(indexPath, content, 'utf-8')
+  // Shape 2 — `defineModules()` fluent chain. Walk forward from
+  // `defineModules(...)` consuming `.mount(...)` calls with a
+  // balanced-paren scanner so nested parens (`mount(UserModule())`)
+  // don't confuse the boundary. We need a real scanner here — a
+  // naive `\.mount\([^)]*\)` regex matches `mount(UserModule()`
+  // and breaks subsequent appends.
+  const chainEnd = findChainEnd(content)
+  if (chainEnd !== -1) {
+    return `${content.slice(0, chainEnd)}\n  .mount(${entryToken})${content.slice(chainEnd)}`
```
[the remark this answers is not in the document]
Comment (forinda) [reply packages/cli/src/generators/module.ts]: @claude Can AST work better than regex in these kind of refactors and code appends ?

**Comment by forinda** [comment]
Defensive escape applied across all four dynamic-regex sites in [`d8e8142`](https://github.com/forinda/kick-js/pull/196/commits/d8e81424).

New `utils/regex.ts` exports a single `escapeRegex(s)` helper used by both `module.ts` (the auto-register name check) and `remove-module.ts` (import-line strip, array-entry strip, chain-mount strip). Agreed it's low-risk today since `pascal` / `plural` flow through `toPascalCase` / `pluralize(toKebabCase())`, but escaping keeps the code correct if upstream sanitization ever loosens — and it's two lines.

No behaviour change. 260/260 still pass.

**Comment by forinda** [comment]
Both addressed in [`3d536de`](https://github.com/forinda/kick-js/pull/196/commits/3d536dea).

| # | Fix |
| - | --- |
| 1 | **Word boundary on the array-entry rm regex.** Added `\b` after `Module` so `UserModule` no longer matches inside `UserModuleFactory`. Brings the array path to parity with `stripChainMount` which already used `\b`. |
| 2 | **Comment-skipping in `balancedClose` / `balancedBracketClose` / the chain-mount scanner.** New `skipComment(src, i)` helper used by both scanners; `remove-module.ts`'s inline scanner got the equivalent. A `)` or `]` inside `//` or `/* ... */` no longer terminates the scan early. |

Two regression tests cover both: `rm` doesn't strip `UserModule` from `UserModuleFactory`, and the chain scanner walks past `/* a comment with ) parens */` without breaking the append.

260 → 262 passing.
